src/applications/vaos/tests/new-appointment/components/ContactInfoPage.unit.spec.jsx

```diff
@@ -1,106 +1,116 @@
-    expect(history.push.called).to.be.false;
-    form.unmount();
+    expect(screen.findByText(/^Please enter a phone number/)).to.be.ok;
+    expect(screen.findByText(/^Please choose at least one option/)).to.be.ok;
+    expect(screen.findByText(/^Please provide a response/)).to.be.ok;
+
+    userEvent.click(button);
+    expect(screen.history.push.called).to.be.false;
   });
 
-  it('should call updateFormData after change', () => {
-    const openFormPage = sinon.spy();
-    const updateFormData = sinon.spy();
-    const history = {
-      push: sinon.spy(),
-    };
-
-    const form = mount(
-      <ContactInfoPage
-        openFormPage={openFormPage}
-        updateFormData={updateFormData}
-        history={history}
-        data={{}}
-      />,
-    );
+  it('should call updateFormData after change', async () => {
+    const store = createTestStore({
+      newAppointment: {
+        pages: [],
+        previousPages: [],
+      },
+    });
+
+    const screen = renderWithStoreAndRouter(<ContactInfoPage />, {
+      store,
+    });
 
-    fillData(form, 'input#root_phoneNumber', '5555555555');
+    // it should display page heading
+    expect(await screen.findByText('Your contact information')).to.be.ok;
 
-    expect(updateFormData.firstCall.args[2].phoneNumber).to.equal('5555555555');
-    form.unmount();
+    const input = document.getElementById('root_phoneNumber');
+    expect(input.value).to.equal('');
+
+    userEvent.type(input, '5555555555');
+
+    expect(input.value).to.equal('5555555555');
   });
 
-  it('should submit with valid data', () => {
-    const openFormPage = sinon.spy();
-    const routeToNextAppointmentPage = sinon.spy();
-
-    const form = mount(
-      <ContactInfoPage
-        openFormPage={openFormPage}
-        routeToNextAppointmentPage={routeToNextAppointmentPage}
-        data={{
-          phoneNumber: '5555555555',
-          email: 'fake@va.gov',
-          bestTimeToCall: {
-            morning: true,
-          },
-        }}
-      />,
+  it('should submit with valid data', async () => {
+    const store = createTestStore({
+      newAppointment: {
+        pages: [],
+        previousPages: [],
+      },
+    });
+
+    const screen = renderWithStoreAndRouter(
+      <Route component={ContactInfoPage} />,
+      {
+        store,
+      },
     );
 
-    form.find('form').simulate('submit');
+    // it should display page heading
+    expect(await screen.findByText('Your contact information')).to.be.ok;
 
-    expect(form.find('.usa-input-error').length).to.equal(0);
-    expect(routeToNextAppointmentPage.called).to.be.true;
-    form.unmount();
-  });
+    let input = document.getElementById('root_phoneNumber');
+    userEvent.type(input, '5555555555');
 
-  it('document title should match h1 text', () => {
-    const openFormPage = sinon.spy();
-    const pageTitle = 'Your contact information';
+    const checkbox = document.getElementById('root_bestTimeToCall_morning');
```

Comment:
Can we get these checkboxes and text fields by their label text, instead of by id?

Reply:
Using getByRole?

Reply:
@vbahinwillit You could use `getByRole` or `getByLabelText`.

I don't think we should stop using `getByRole` altogether, we should just be wary of using on pages that have lots of elements with the role we're searching for. This page is pretty light on form elements.

Reply:
Done.